packages/buffetjs-core/src/components/DatePicker/index.js

```diff
@@ -35,107 +51,105 @@ class DatePicker extends React.PureComponent {
+  const handleDateClick = date => {
+    handleDateChange(date);
+
+    toggleDatepicker(false);
   };
 
-  handleOutsideClick = () => {
-    this.setState({
-      visible: false,
-    });
+  const handleOutsideClick = () => {
+    toggleDatepicker(false);
   };
 
-  handleTabClick = ({ keyCode, which }) => {
+  const handleTabClick = ({ keyCode, which }) => {
     const code = keyCode || which;
 
     if (code === 9) {
-      this.handleOutsideClick();
+      handleOutsideClick();
     }
   };
 
-  showDatepicker = () => {
-    this.setState({
-      visible: true,
-    });
+  const toggleDatepicker = shown => {
+    setState(prevState => ({
+      ...prevState,
+      visible: shown,
+      isFocused: shown,
+    }));
   };
 
-  render() {
-    const {
-      className,
-      disabled,
-      displayFormat,
-      id,
-      name,
-      readOnly,
-      tabIndex,
-    } = this.props;
-    const { date, displayedDate, isFocused, visible } = this.state;
-    let dateValue = date ? moment(date).format(displayFormat) : '';
-
-    if (visible) {
-      dateValue = displayedDate;
-    }
-
-    return (
-      <StyledDatepicker className={`${className} ${visible && 'is-open'}`}>
-        <div>
-          <Input
-            disabled={disabled}
-            type="text"
-            name="start_date"
-            id={id || name}
-            value={dateValue}
-            readOnly={readOnly}
-            onChange={this.handleChange}
-            icon={<FontAwesomeIcon icon={faCalendarAlt} />}
-            onClick={this.showDatepicker}
-            onFocus={this.showDatepicker}
-            onKeyDown={this.handleTabClick}
-            tabIndex={tabIndex}
-          />
-        </div>
-        {visible && (
-          <DayPickerSingleDateController
-            date={date}
-            focused={isFocused}
-            numberOfMonths={1}
-            keepOpenOnDateSelect
-            onFocusChange={this.handleFocusChange}
-            onDateChange={this.handleDateChange}
-            onOutsideClick={this.handleOutsideClick}
-          />
-        )}
-      </StyledDatepicker>
-    );
-  }
+  return (
+    <StyledDatepicker className={`${className} ${state.visible && 'is-open'}`}>
+      <div>
+        <Input
+          disabled={disabled}
+          type="text"
+          name="start_date"
+          id={id || name}
+          value={getDateValue()}
+          readOnly={readOnly}
+          onChange={handleChange}
+          icon={<FontAwesomeIcon icon={faCalendarAlt} />}
+          onClick={() => toggleDatepicker(true)}
+          onKeyDown={handleTabClick}
+          tabIndex={tabIndex}
+        />
+      </div>
+      {state.isFocused && (
+        <DayPickerSingleDateController
+          date={state.date}
+          focused
+          numberOfMonths={1}
+          onDateChange={handleDateClick}
+          onOutsideClick={handleOutsideClick}
+          daySize={37}
+          transitionDuration={0}
+        />
```

Comment:
```suggestion
          renderMonthElement={({ month, onYearSelect }) => (
            (state.yearSelect ?
              <select
                value={month.year()}
                onChange={(e) => {
                  onYearSelect(month, e.target.value);
                  toggleYearSelect();
                }}
              >
                {range(month.year() - 7, month.year() + 7)
                  .map((option, index) => <option key={index.id} value={option}>{option}</option>)}
              </select>
              :
              <strong onClick={toggleYearSelect}>{month.format('MMMM YYYY')}</strong>
            )
          )}
        />
```

``` 
import {range} from 'lodash'

...

  const [state, setState] = useState({
    ...
    yearSelect: false,
  });

...

  const toggleYearSelect = () => {
    setState(prevState => ({
      ...prevState,
      yearSelect: !state.yearSelect,
    }));
  };

```

An idea for the feature-request discussed here: https://strapi.slack.com/archives/C0BNGCCS1/p1580916261047900

Reply:
Hi!

Thank you very much for your review and your suggestions.

Your comment is very relevant. However, we need some design review for this feature so can you please open another PR with this suggested change? Like that we can test and review it easily.

Thanks!

Reply:
Hey,

Thanks for applying my suggestion. 
Yes of course, I'll create a new PR with a concept for the suggestion.